### PR TITLE
Fun With Flags

### PR DESCRIPTION
**Pull request by tebeka**
Article about the `flag` package

### CODE REVIEW

File: content/advent-2019/flags.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+	*p.port = val
+	return nil
+}
+```
+
+## Sub Commands
+
+Instead of having one executable to start the HTTP server and another to check
+it's alive. We can have one executable that does both commands (same as `git`
```
Comment (Pisush) [inline content/advent-2019/flags.md]: This seems like it should be a comma connecting the two sentences?

File: content/advent-2019/flags.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+```
+
+1. Usage for the main executable
+2. We should have at lest two `os.Args` - the executable and the sub command name
+3. `os.Args[1]` is the subcommand (`os.Args[1]` is the executable name)
+4. Create a new `FlagSet` to parse the command line for this sub command. Use
+   `flag.ContinueOnError` so parse error will not exit the program. The only function
+   that should exit the program is `main`, all others should return an error
+5. Pass rest of arguments. e.g. `["app", "check", "http://localhost:8080"]` →
+   `["localhost:8081]`
```
Comment (Pisush) [inline content/advent-2019/flags.md]: missing a closing `"`

File: content/advent-2019/flags.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+You can see the full source code for the examples
+[here](https://github.com/gopheracademy/gopheracademy-web/blob/master/content/advent-2019/flag).
+
+
+# About the Author
+Hi there, I'm Miki, nice to e-meet you ☺. I've been a long time developer and
+have been working with Go for about 10 years. I write code professionally as
+a consultant and contribute a lot to open source. Apart from that I'm a [book
+author](https://www.amazon.com/Forging-Python-practices-lessons-developing-ebook/dp/B07C1SH5MP) author, an author on [LinkedIn
```
Comment (Pisush) [inline content/advent-2019/flags.md]: double `author`

File: content/advent-2019/flags.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+learning](https://www.linkedin.com/learning/search?keywords=miki+tebeka), one of
```
Comment (Pisush) [inline content/advent-2019/flags.md]: Not-a-deal-breaker suggestion for another link: https://www.linkedin.com/learning/instructors/miki-tebeka

File: content/advent-2019/flags.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
++++
+title = "Fun With Flags"
+date = "2019-12-08T00:00:00+00:00"
```
[the remark this answers is not in the document]
Comment (tebeka) [reply content/advent-2019/flags.md]: Yeah, my bad. Fixed

File: content/advent-2019/flags.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+series = ["Advent 2019"]
+author = ["Miki Tebeka"]
++++
+
+In a [previous article](FIXME) we discussed why command line applications are
```
[the remark this answers is not in the document]
Comment (tebeka) [reply content/advent-2019/flags.md]: Fixed.

File: content/advent-2019/flags.md (hunk 1 above)
Comment (tebeka) [reply content/advent-2019/flags.md]: Yup. Fixed.

File: content/advent-2019/flags.md (hunk 3 above)
Comment (tebeka) [reply content/advent-2019/flags.md]: Thanks. You're the first out of 3 reviewers to catch this.

File: content/advent-2019/flags.md (hunk 2 above)
Comment (tebeka) [reply content/advent-2019/flags.md]: Fixed.

File: content/advent-2019/flags.md (hunk 4 above)
Comment (tebeka) [reply content/advent-2019/flags.md]: Even better.

**Comment by tebeka** [comment]
Thanks @Pisush !

**Pull request merged**